[SP-1855]PDI-13759] excel input step when using apache POI streaming …

SergeyTravin opened the pull request:
…does not release file xlsx in case of error

SergeyTravin commented:
@mattcasters, @brosander, @mattyb149 Could you review it, please? Thanks!
It is backport for https://github.com/pentaho/pentaho-kettle/pull/1321

SergeyTravin commented:
@mattcasters, @brosander, @mattyb149 
I would be grateful if you could review the PR before end of the July SP.
Thanks a lot!

brosander commented:
@SergeyTravin why are we manually catching exceptions and then failing with the message?  If you just add a throws declaration to the method signature, the test will show an error if it gets an unexpected exception

SergeyTravin commented:
@brosander, I removed unnecessary catch's block and added  throws declaration to the method signature.  It seems pretty good now. Could you process it?

brosander commented on engine/src/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxPoiWorkbook.java:
```diff
@@ -125,13 +132,19 @@ public KSheet getSheet( String sheetName ) {
+        //We should not save change in xlsx because it is input step.
+        opcpkg.revert();
+      }
+    } catch ( IOException e ) {
```
@SergeyTravin Shouldn't we be doing the try/catch inside the for loop so we still try to close all sheets even if one fails to close?

SergeyTravin commented:
@brosander, updated PR according your notes. Could you look at it again?
Great thanks!

SergeyTravin commented:
@hudak, @mattcasters, @brosander, @mattyb149, Could you review this backport for https://github.com/pentaho/pentaho-kettle/pull/1321? This issue is included to the July SP that finishes on Friday.
/cc @mbradbury2014, @dkincade

SergeyTravin commented:
@hudak, @mattcasters, @brosander, @mattyb149, Could you review this backport for #1321? This issue is included to the July SP that finishes on Friday.
/cc @mbradbury2014, @dkincade
